docs/user/unsupported/mbstyle.rst

```diff
@@ -19,14 +24,544 @@ Quick example of module usage (API change):
+
+    ["<=", *key, value*] less than or equal: *feature[key] ≤ value*
+
+    **Set Membership Filters**
+
+    ["in", *key, v0, ..., vn*] set inclusion: *feature[key] ∈ {v0, ..., vn}*
+
+    ["!in", *key, v0, ..., vn*] set exclusion: *feature[key] ∉ {v0, ..., vn}*
+
+    **Combining Filters**
+
+    ["all", *f0, ..., fn*] logical AND: *f0 ∧ ... ∧ fn*
+
+    ["any", *f0, ..., fn*] logical OR: *f0 ∨ ... ∨ fn*
+
+    ["none", *f0, ..., fn*] logical NOR: *¬f0 ∧ ... ∧ ¬fn*
+
+    A *key* must be a string that identifies a feature property, or one of the following special keys:
+
+      * "$type": the feature type. This key may be used with the "==",  "!=", "in", and "!in" operators. Possible values are  "Point", "LineString", and "Polygon".
+
+      * "$id": the feature identifier. This key may be used with the "==",  "!=", "has", "!has", "in", and "!in" operators.
+
+    A *value* (and *v0, ..., vn* for set operators) must be a string, number, or boolean to compare the property value against.
+
+    Set membership filters are a compact and efficient way to test whether a field matches any of multiple values.
+
+    The comparison and set membership filters implement strictly-typed comparisons; for example, all of the following evaluate to false: 0 < "1", 2 == "2", "true" in [true, false].
+
+    The "all", "any", and "none" filter operators are used to create compound filters. The values *f0, ..., fn* must be filter expressions themselves.
+
+    .. code-block:: json
+
+      ["==", "$type", "LineString"]
+
+    This filter requires that the class property of each feature is equal to either "street_major", "street_minor", or "street_limited".
+
+    .. code-block:: json
+
+      ["in", "class", "street_major", "street_minor", "street_limited"]
+
+    The combining filter "all" takes the three other filters that follow it and requires all of them to be true for a feature to be included: a feature must have a class equal to "street_limited", its admin_level must be greater than or equal to 3, and its type cannot be Polygon. You could change the combining filter to "any" to allow features matching any of those criteria to be included - features that are Polygons, but have a different class value, and so on.
+
+    .. code-block:: json
+
+      [
+        "all",
+        ["==", "class", "street_limited"],
+        [">=", "admin_level", 3],
+        ["!in", "$type", "Polygon"]
+      ]
+
+MapBox Style Grammar
+^^^^^^^^^^^^^^^^^^^^
+  *JSON does not allow for comments within the data therefore comments will be noted through the placement of the comment between open < and close > angle brackets. All properties are optional unless otherwise noted as Requried*
+
+  Root Properties
+  '''''''''''''''
+
+  .. code-block:: json
+
+    {
+      "version": 8, <Required>
+      "name": "Mapbox Streets",
+      "sprite": "mapbox://sprites/mapbox/streets-v8",
+      "glyphs": "mapbox://fonts/mapbox/{fontstack}/{range}.pbf",
+      "sources": {...}, <Required>
```

Comment:
While `sources` is required by the spec, the current MBStyle geotools implementation does not enforce this. Maybe add a note to this effect?